src/libs/print_settings.c

```diff
@@ -0,0 +1,1168 @@
+  dt_conf_set_string("plugins/print/print/iccprofile", "image");
+}
+
+static void
+_printer_profile_changed(GtkComboBox *widget, dt_lib_module_t *self)
+{
+  dt_lib_print_settings_t *ps = (dt_lib_print_settings_t *)self->data;
+  int pos = gtk_combo_box_get_active(widget);
+  GList *prof = ps->profiles;
+  while(prof)
+  {
+    // could use g_list_nth. this seems safer?
+    dt_lib_export_profile_t *pp = (dt_lib_export_profile_t *)prof->data;
+    if(pp->pos == pos)
+    {
+      dt_conf_set_string("plugins/print/printer/iccprofile", pp->filename);
+      return;
+    }
+    prof = g_list_next(prof);
+  }
+  dt_conf_set_string("plugins/print/printer/iccprofile", "image");
+}
+
+static void
+_printer_intent_callback (GtkComboBox *widget, dt_lib_module_t *self)
+{
+  int pos = gtk_combo_box_get_active(widget);
+  dt_conf_set_int("plugins/print/printer/iccintent", pos);
+}
+
+static void
+_intent_callback (GtkComboBox *widget, dt_lib_module_t *self)
+{
+  int pos = gtk_combo_box_get_active(widget);
+  // record the intent that will override the out rendering module on export
+  dt_conf_set_int("plugins/lighttable/export/iccintent", pos - 1);
+  dt_conf_set_int("plugins/print/print/iccintent", pos - 1);
+}
+
+static void _print_settings_filmstrip_activate_callback(gpointer instance,gpointer user_data)
+{
+  dt_lib_module_t *self = (dt_lib_module_t *)user_data;
+  dt_lib_print_settings_t *ps = (dt_lib_print_settings_t *)self->data;
+
+  if (ps->prt.page.landscape)
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(ps->landscape), TRUE);
+  else
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(ps->portrait), TRUE);
+}
+
+void
+gui_init (dt_lib_module_t *self)
+{
+  dt_lib_print_settings_t *d = (dt_lib_print_settings_t*)malloc(sizeof(dt_lib_print_settings_t));
+  self->data = d;
+  self->widget = gtk_table_new(18, 2, FALSE);
+  GtkWidget *label;
+  GtkComboBox *comb;
+  int tpos = 1; // the ligne on the table
+  char tooltip[1024];
+
+  gtk_table_set_row_spacings(GTK_TABLE(self->widget), 5);
+
+  dt_init_print_info(&d->prt);
+  d->prt.page.landscape = TRUE;
+  dt_view_print_settings(darktable.view_manager, &d->prt);
+
+  dt_control_signal_connect(darktable.signals,
+                            DT_SIGNAL_VIEWMANAGER_FILMSTRIP_ACTIVATE,
+                            G_CALLBACK(_print_settings_filmstrip_activate_callback),
+                            self);
+
+  //  create the spin-button now as values could be set when the printer has no hardware margin
+
+  d->b_top    = gtk_spin_button_new_with_range(0, 10000, 1);
+  d->b_left   = gtk_spin_button_new_with_range(0, 10000, 1);
+  d->b_right  = gtk_spin_button_new_with_range(0, 10000, 1);
+  d->b_bottom = gtk_spin_button_new_with_range(0, 10000, 1);
+
+  ////////////////////////// PRINTER SETTINGS
+
+  // create papers combo as filled when adding printers
+  d->papers = GTK_COMBO_BOX(gtk_combo_box_text_new());
+
+  GList *printers = dt_get_printers();
```

Comment:
I do not see where this is being freed (what was allocated in dt_get_printer_info())